pkg/operators/operators.go

```diff
@@ -71,6 +81,45 @@ type Operator interface {
 	Instantiate(gadgetContext GadgetContext, gadgetInstance any, params *params.Params) (OperatorInstance, error)
 }
 
+type ImageOperator interface {
+	Name() string
+
+	// InstantiateImageOperator will be run to load information about a gadget and also to _possibly_
+	// run the gadget afterward. It should only do things that are required to populate
+	// DataSources and Params. It could use caching to speed things up, if necessary.
+	InstantiateImageOperator(GadgetContext, ocispec.Descriptor) (ImageOperatorInstance, error)
+}
+
+type ImageOperatorInstance interface {
+	Name() string
+	Prepare(ctx GadgetContext) error
+	Start(ctx GadgetContext) error
+}
+
+type DataOperator interface {
+	Name() string
+
+	// Init allows the operator to initialize itself
+	Init(params *params.Params) error
+
+	// GlobalParamDescs will return global params (required) for this operator
+	GlobalParamDescs() params.ParamDescs
+
+	// InstantiateDataOperator should create a new (lightweight) instance for the operator that can read/write
+	// from and to DataSources, register Params and read/write Variables
+	InstantiateDataOperator(GadgetContext) (DataOperatorInstance, error)
+}
+
+type DataOperatorInstance interface {
+	Name() string
+	Prepare(ctx GadgetContext) error
+	Start(ctx GadgetContext) error
+}
```

Comment:
I think we need to add a `Stop()` method to this interface, it'll help operators that need to perform some clean up. I think the same applies to `ImageOperatorInstance`

Reply:
Yeah, I think you're right - I first thought it would be enough to have a `Start()`, because there you could also just wait for `ctx.Context().Done()` - and less is usually more. But having a dedicated method makes things more clear and deterministic.